Test.C

```diff
@@ -129,76 +138,107 @@ int main(){
                     // check if encryption method effectiveness is greater than the previous max
-                    if (encryptionPtr->encryption.effectiveness > encryptionEffectiveness){
-                        encryptionEffectiveness = encryptionPtr->encryption.effectiveness;
+                    if (encryptionPtr->encryption->effectiveness > encryptionEffectiveness){
+                        encryptionEffectiveness = encryptionPtr->encryption->effectiveness;
                     }
                 }
 
                 encryptionPtr = encryptionPtr->next;
             }
             newCredits += equipmentStock[i].messageAmount * player.equipmentInventory[i] * encryptionEffectiveness; // sum message amount of equipment(s)
         }
-
         player.currentCredits += newCredits; // adding credits to currentCredits
 
         // tell user how credits received and what encrypt method was used
 	      printf("You received %d credits!\n", newCredits); // telling the user how many credits they were awarded
 
         printf("Credits: %d\n", player.currentCredits); // current credit
+        printf("\n");
 
         // Present attack
         // print attack details and affects of attack
         // edit encrypt structure effectiveness
+        if (rounds[currentRound].attack > -1){
+            Attack currentAttack = attackStock[rounds[currentRound].attack];
+            printf("|----------------------------- Warning New Attack! ----------------------------|\n");
+            printf("%s was developed!\n", currentAttack.name);
+            printf("%s\n", currentAttack.description);
+            printf("\n");
+            printf("Affects on Encryption Methods:\n");
+
+            //adjust encryption methods based on attack
+            for (int i=0; i<NUM_ENCRYPTION; i++){
+                double affect = attackEffectiveness[i][rounds[currentRound].attack];
+                if (affect > 0){
+                    encryptionStock[i].effectiveness -= affect;
+                    // keep in bounds
+                    if (encryptionStock[i].effectiveness < 0)
+                        encryptionStock[i].effectiveness = 0;
+                    printf(" - %s effectiveness has been reduced to %.2f!\n", encryptionStock[i].name, encryptionStock[i].effectiveness);
+                }
+            }
+        }
+        else{
+            printf("|------------------------------- No New Attack! -------------------------------|\n");
+        }
+
+
+        printf("\n");
 
         // Spend credits interactive menu
         printf("|-------------------------------- Credit Store --------------------------------|\n");
         printf("Here you can buy equipment and upgrade encryption as well as sell the equipment\n");
         printf("that you already own.\n");
-        printf("Your current credits are: %d\n", player.currentCredits); //current credit
+        printf("\n");
```

Comment:
```suggestion
        printf("Your current credits are: %d\n", player.currentCredits); //current credit
        printf("\n");
```

Reply:
I think it's important to include how many credits the user has to spend so they are aware. I'm open to including it somewhere else though.